oeps/oep-0028-content-theming-in-react.rst

```diff
@@ -0,0 +1,246 @@
+            return (
+                <MainNavWrapper>
+                    <a href="/">Home</a>
+                    <LoginLink/>
+                    {this.extraNavLinks}
+                </MainNavWrapper>
+            );
+        }
+        get extraNavLinks() { return []; }
+    }
+    // Internal MainNavWrapper - not meant to be modified in most cases
+    class _MainNavWrapper extends React.PureComponent {
+        render() {
+            return (
+                <div className="mainNav">
+                    <ul>
+                        {React.Children.map(this.props.children, (child) => (child ? <li>{child}</li> : null))}
+                    </ul>
+                </div>
+            )
+        }
+    }
+
+    // Default Theme:
+    export const Header = _Header;
+    export const MainNav = _MainNav;
+    export const MainNavWrapper = _MainNavWrapper;
+
+
+Now if we want to customize our _Header component, we can easily do it like
+
+.. code-block:: js
+
+    // SiteLogo being updated in Header
+    class MyThemedHeader extends _Header {
+        render() {
+            return (
+                <header>
+                    {/* Replace <SiteLogo/> with a fancy widget */}
+                    <MyCustomAnimatedLogoWidget/>
+                    <MainNav/>
+                    <UserAvatar/>
+                </header>
+            );
+        }
+    }
+    // Custom theme:
+    export const Header = MyThemedHeader;
+    export const MainNavWrapper = _MainNavWrapper;
+
+
+* We will provide support to add data or other components by using `props.children`. The `props.children` property is provided from React and can be extended to include specialized children like left and right for a Pane component. An example of this can be
+
+.. code-block:: js
+
+    const Header = (props) => {
+        return (
+            <h1>{props.title}</h1>
+            {props.children}
+        )
+    }
+
+    // use
+    <Header title={"Open Edx"} >
+        <h4>Open Source MOOC platform</h4>
+    </Header>
+
+Another example where specialized children are passed
+
+.. code-block:: js
+
+    const Pane = (props) => {
+        return (
+            <div className="pane">
+                <div className="pane-left">
+                    {props.left}
+                </div>
+                <div className="pane-right">
+                    {props.right}
+                </div>
+            </div>
+
+        )
+    }
+
+    // use
+    <Pane left={<Image />} right={<Explanation />} />
+
+
+* We will use function and placeholder to add additional content in customizable components. We can take an example of the above DefaultTheme and see _MainNav where it has support to add additional nav links by overriding `extraNavLinks` function.
+
+.. code-block:: js
+
+    // Customizable Main Navigation Area
+    class MyThemedNav extends _MainNav {
+        get extraNavLinks() {
+            return [
+                <a href="/about">About Us</a>,
+            ];
+        }
```

Comment:
I think a better approach is to use [React fragments](https://reactjs.org/docs/fragments.html) here. So it could be:
```javascript
return (
    <Fragment>
        <a href="/about">About Us</a>,
    <Fragment>
);
```
or
```javascript
return (
    <>
        <a href="/about">About Us</a>,
    </>
);
```

Reply:
Arrays are just a way of returning multiple elements, and that is now officially supported by React, [Fragments are a replacement for return an array](https://reactjs.org/blog/2017/11/28/react-v16.2.0-fragment-support.html).

Thanks for pointing out the comma issue, you're right that it would be rendered in this case, I just forgot to clean them up while copy-pasting from the existing code.

Reply:
I have updated the snippet to use `React.Fragments`. Thanks :)